### PR TITLE
Fix an issue with an incorrect retry default timeout for urllib3 1.x

### PR DESCRIPTION
**Pull request by BenWilson2**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/BenWilson2/mlflow/pull/10839?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10839/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10839
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Fixes an issue where a retry with jittered delay would pull the incorrect constant that had been renamed in urllib3 2.x. This fix handles the renaming resolution. 

### How is this PR tested?

- [X] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [X] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [X] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by B-Step62 — APPROVED** [review]
LGTM!

File: mlflow/utils/request_utils.py [hunk 1]
```diff
@@ -43,7 +43,16 @@ def get_backoff_time(self):
...
             backoff_value += random.random() * self.backoff_jitter
-        return float(max(0, min(Retry.DEFAULT_BACKOFF_MAX, backoff_value)))
+        # The attribute `BACKOFF_MAX` was renamed to `DEFAULT_BACKOFF_MAX` in this commit:
+        # https://github.com/urllib3/urllib3/commit/f69b1c89f885a74429cabdee2673e030b35979f0
+        # which was part of the major release of 2.0 for urllib3
+        default_backoff = (
+            Retry.BACKOFF_MAX
+            if Version(urllib3.__version__) < Version("2.0")
+            else Retry.DEFAULT_BACKOFF_MAX
+        )
```
Comment (harupy) [inline mlflow/utils/request_utils.py:53]: if urllib3_version < Version("2.0"):
        retry = JitteredRetry(**retry_kwargs)
    else:
        retry = Retry(**retry_kwargs)

Comment (harupy) [reply mlflow/utils/request_utils.py:53]: Wondering why we hit this issue. `Retry.DEFAULT_BACKOFF_MAX` is valid in urllib3 < 2.0.

Comment (BenWilson2) [reply mlflow/utils/request_utils.py:53]: I'll remove the version check - thanks for pointing it out :) 
As for the Retry.DEFAULT_BACKOFF_MAX in versions prior to 2.0, I can't find its existence in their repo as of this parent to the commit referenced in this PR: https://github.com/urllib3/urllib3/blob/c96cf403fb4f24d414f40faf4691174e4c54ea0b/src/urllib3/util/retry.py

Comment (harupy) [reply mlflow/utils/request_utils.py:53]: I'm also wondering where the version boundary is.

**Review by harupy — APPROVED** [review]
LGTM

**Pull request merged**